document/core/exec/runtime.rst

```diff
@@ -277,24 +315,26 @@ It is an invariant of the semantics that the length of the element vector never
 Memory Instances
 ~~~~~~~~~~~~~~~~
 
+.. todo:: We used to update the memory type when a memory is grown. This does not work for shared memories. In fact, the "current" type of a shared memory is nondeterministic. We need to model that during instantiation somehow.
```

Comment:
I had assumed that instantiation would proceed as though executing `memory.size` on each imported memory (similar to the way global initialisers are evaluated), but I realise now that this hasn't really been discussed.

IIUC, the memory type is only introspected during instantiation, so for instruction validation and the soundness statement, it would be OK for the memory type to no longer contain a `min` size?

Reply:
Memory types are used for a number of purposes, and its notion of limits is furthermore shared with tables. I'd rather not introduce multiple different variations and intersections of these concepts. Rather, some form of abstraction to read the current "dynamic" type of a memory at instantiation time. That would also be what's used by the embedder API.

Reply:
to be bikeshedded when instantiation is specified